staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -104,6 +104,7 @@ func PatchResource(r rest.Patcher, scope RequestScope, admit admission.Interface
 		result, err := patchResource(ctx, updateAdmit, timeout, versionedObj, r, name, patchType, patchJS,
 			scope.Namer, scope.Creater, scope.Defaulter, scope.UnsafeConvertor, scope.Kind, scope.Resource, codec)
 		if err != nil {
+			err = errors.NewBadRequest(err.Error())
```

Comment:
what if there is a sensible http-compatible error from patchResource? Like NotFound.